src/com/phonebookservice/database/ContactsDatabase.java

```diff
@@ -0,0 +1,66 @@
+package com.phonebookservice.database;
+
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+
+import com.phonebookservice.model.Contact;
+import com.phonebookservice.util.MyArrayList;
+
+public final class ContactsDatabase {
+    private static ContactsDatabase singleton;
+    private final ContactsFileParser contactsFileParser;
+    private final Collection<Contact> contactList;
+    private final Map<Long, Contact> idToContactMap;
+
+    /**
+     * Contacts database constructor.
+     *
+     * @param fileName the file name.
+     */
+    private ContactsDatabase(final String fileName) {
+        this.idToContactMap = new HashMap<>();
+        this.contactsFileParser = new ContactsFileParser(fileName);
+        this.contactList = //
+                new MyArrayList<Contact>(contactsFileParser.readContacts());
+        addInContactMap();
```

Comment:
I suggest to Fill the array and Hashtable in same place 

Array = parser.read()

Loop (contact in array )
   Array.add(contact)
   Hashtable.add(id , contact)